File: go/gomod/adapter/hardcoded_catalog.go
package adapter

import "example.com/catalog/core"

type HardcodedCatalog struct {}

func (adapter HardcodedCatalog) CatalogItems() []core.CatalogItem {
	return []core.CatalogItem {
		core.CatalogItem {
			Name:         "Courage",
			PriceInCents: 10_00,
		},
		core.CatalogItem {
			Name:         "Brain",
			PriceInCents: 2,
		},
		core.CatalogItem {
			Name:         "Heart",
			PriceInCents: 7_89,
		},
	}
}

File: ruby/bundler/adapter/lib/adapter/hardcoded_catalog.rb
require 'core/catalog_item'

module Adapter
  class HardcodedCatalog
    def catalog_items
      [
        Core::CatalogItem.new("Courage", 10_00),
        Core::CatalogItem.new("Brain", 2),
        Core::CatalogItem.new("Heart", 7_89),
      ]
    end
  end
end